features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapListener.java

```diff
@@ -94,6 +99,17 @@ public void trapError(int error, String msg) {
     }
 
     public void start() {
+        CompletableFuture<OnmsTwin> future = onmsTwinSubscriber.getObject("trapd-config", new OnmsTwinSubscriber.SinkCallback() {
```

Comment:
Can we make it so that we get the `TrapdConfig` object back in a typesafe fashion and not have to unmarshal it ourselves?

Reply:
I can change this to get `TrapdConfig` object by passing specific Class.

Reply:
Updated API with specific types.